src/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsType.php

```diff
@@ -79,35 +81,103 @@ public function __construct(
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('enable_invoices', SwitchType::class)
-            ->add('enable_tax_breakdown', SwitchType::class)
-            ->add('enable_product_images', SwitchType::class)
+            ->add('enable_invoices', SwitchType::class, [
+                'label' => $this->trans('Enable invoices', 'Admin.Orderscustomers.Feature'),
+                'help' => $this->trans('If enabled, your customers will receive an invoice for the purchase.', 'Admin.Orderscustomers.Help'),
+            ])
+            ->add('enable_tax_breakdown', SwitchType::class, [
+                'label' => $this->trans('Enable tax breakdown', 'Admin.Orderscustomers.Feature'),
+                'help' => $this->trans('If required, show the total amount per rate of the corresponding tax.', 'Admin.Orderscustomers.Help'),
+            ])
+            ->add('enable_product_images', SwitchType::class, [
+                'label' => $this->trans('Enable product image', 'Admin.Orderscustomers.Feature'),
+                'help' => $this->trans('Add an image in front of the product name on the invoice.', 'Admin.Orderscustomers.Help'),
+            ])
             ->add('invoice_prefix', TranslatableType::class, [
+                'options' => [
+                    'constraints' => [
+                        new NoTags(),
+                    ],
+                ],
+                'required' => false,
                 'type' => TextType::class,
+                'label' => $this->trans('Invoice prefix', 'Admin.Orderscustomers.Feature'),
+                'help' => $this->trans('Freely definable prefix for invoice number (e.g. #IN00001).', 'Admin.Orderscustomers.Help'),
+            ])
+            ->add('add_current_year', SwitchType::class, [
+                'label' => $this->trans('Add current year to invoice number', 'Admin.Orderscustomers.Feature'),
+            ])
+            ->add('reset_number_annually', SwitchType::class, [
+                'label' => $this->trans('Reset sequential invoice number at the beginning of the year', 'Admin.Orderscustomers.Feature'),
             ])
-            ->add('add_current_year', SwitchType::class)
-            ->add('reset_number_annually', SwitchType::class)
             ->add('year_position', ChoiceType::class, [
+                'required' => false,
                 'choices' => [
                     $this->trans('After the sequential number', 'Admin.Orderscustomers.Feature') => 0,
                     $this->trans('Before the sequential number', 'Admin.Orderscustomers.Feature') => 1,
                 ],
                 'expanded' => true,
+                'label' => $this->trans('Position of the year date', 'Admin.Orderscustomers.Feature'),
             ])
             ->add('invoice_number', NumberType::class, [
                 'required' => false,
+                'constraints' => [
```

Comment:
I think we could use this opportunity to add better wording here. Currently it checks if invoice number is larger then 0 and if it's larger then last invoice number. Maybe error message should reflect that instead of plain "Invalid invoice number".

Reply:
@Julievrz @LouiseBonnard what do you think of this idea?

Today there is a little logic that says "Invalid invoice number"
- if submitted invoice number is below zero (obviously wrong)
- if submitted invoice number is too high, greater than last invoice number

Do you think you could give us 2 wordings, one for each case, so it's more accurate for merchant?

Reply:
Hi! I agree that the wording should be more specific. However, I am not sure to understand the 2nd point. Indeed, I have tried to write "3000" for the invoice number. When I download the invoice the number is 3000. Then, I tried to set "4000" and there was no error message even though 4000 is greater than 3000 (the last invoice number)

Reply:
Slight correction.
Error is thrown if
1. Invoice number is below zero(correct).
2. Invoice number is not greater then last invoice number. So in your example @Julievrz  now that you changed invoice number to 4000 and created the invoice you shouldn't be be able to set this setting to 4000 or anything less then that.

Reply:
I suggest "Invoice number is invalid. Please enter a positive integer, greater than the last invoice number."

Reply:
Or even better, "Invoice number is invalid. Please enter a positive integer greater than the last invoice number (#%number%)." in Admin.Orderscustomers.Notification.

Wdyt? @JevgenijVisockij

Reply:
Yeah, makes sense to me!